services/audit-service/src/main/java/datawave/microservice/audit/controller/AuditController.java

```diff
@@ -105,25 +105,32 @@ public void processConfirmAck(Message<?> message) {
      * @param parameters
      *            The audit parameters to be sent
      */
-    private void sendMessage(AuditParameters parameters) {
-        String auditId = parameters.getAuditId();
-        
-        CountDownLatch latch = null;
-        if (auditProperties.isConfirmAckEnabled()) {
-            latch = new CountDownLatch(1);
-            correlationLatchMap.put(auditId, latch);
+    private boolean sendMessage(AuditParameters parameters) {
+        if ((healthChecker != null && healthChecker.isHealthy()) || healthChecker == null) {
+            String auditId = parameters.getAuditId();
+            
+            CountDownLatch latch = null;
+            if (auditProperties.isConfirmAckEnabled()) {
+                latch = new CountDownLatch(1);
+                correlationLatchMap.put(auditId, latch);
+            }
+            
+            boolean success = messageChannel.send(MessageBuilder.withPayload(AuditMessage.fromParams(parameters)).setCorrelationId(auditId).build());
+            
+            if (success && auditProperties.isConfirmAckEnabled()) {
+                try {
+                    success = latch.await(auditProperties.getConfirmAckTimeoutMillis(), TimeUnit.MILLISECONDS);
+                } catch (InterruptedException e) {
+                    success = false;
+                } finally {
+                    correlationLatchMap.remove(auditId);
```

Comment:
If success is false above, then you'll never remove auditId from the map.

Reply:
Also, not sure if this is a problem or not, but it seems like there might be a race condition where a slow confirm from one audit call to count down the latch of a previous timed out call. That would be a duplicate anyway, so that's why I'm thinking it might not be a problem.

Reply:
That's a good catch.  I'll throw out a fix tomorrow morning. 

W.R.T. duplicates, that's definitely possible if the confirm ever takes longer than the confirmAckTimeout.  We definitely open ourselves up to that by allowing for retry logic.  I'm not sure how to better handle that case...

Reply:
@brianloss I think I addressed your concerns with the latest commit.